invokeai/frontend/web/src/features/deleteImageModal/components/DeleteImageButton.tsx

```diff
@@ -13,13 +13,18 @@ export const DeleteImageButton = memo((props: DeleteImageButtonProps) => {
   const { onClick, isDisabled } = props;
   const { t } = useTranslation();
   const isConnected = useAppSelector((s) => s.system.isConnected);
+  const imageSelectionLength: number = (useAppSelector((s) => s.gallery.selection) || []).length;
```

Comment:
When using selectors with redux, it's usually preferred to move computation inside the selector and always preferred to return a primitive value. This allows the built-in memoization to prevent component re-renders.

In this case, it's a single button, so no big deal, but it's cleaner to do this anyways.

```suggestion
  const imageSelectionLength = useAppSelector((s) => s.gallery.selection.length);
```

Reply:
Makes much sense. Thanks for the insight.